src/lib.rs

```diff
@@ -99,10 +102,12 @@ impl Default for Client {
 
         #[cfg(all(feature = "rustls-tls", not(feature = "rustls-tls-aws")))]
         let connector =
-            prepare_hyper_rustls_client(connector, rustls::crypto::ring::default_provider());
+            prepare_hyper_rustls_connector(connector, rustls::crypto::ring::default_provider());
         #[cfg(all(feature = "rustls-tls-aws", not(feature = "rustls-tls")))]
```

Comment:
These cfg should include `not(feature = "native-tls")` otherwise having rustls & native-tls enabled at same time breaks

Reply:
Thinking over, I have an idea that'd be clear:
Use cfg!() if-else for the 4 cases in order of priority. Then have their init functions defined to be stubs when the feature is not enabled

Reply:
#141